### PR TITLE
Performance tool: Warn when interrupting an in-progress tree

### PR DESCRIPTION
**Pull request by acdlite**
Adds a warning to the performance tool when an in-progress update is interrupted by equal or higher priority work.

### CODE REVIEW

File: packages/react-reconciler/src/ReactDebugFiberPerf.js [hunk 1]
```diff
@@ -332,14 +333,17 @@ export function startWorkLoopTimer(): void {
...
     }
-    const warning = commitCountInCurrentWorkLoop > 1
-      ? 'There were cascading updates'
-      : null;
+    let warning = null;
+    if (didInterrupt) {
+      warning = 'Interrupted an in-progress update';
+    } else if (commitCountInCurrentWorkLoop > 1) {
+      warning = 'There were cascading updates';
+    }
```
Comment (acdlite) [inline packages/react-reconciler/src/ReactDebugFiberPerf.js]: @gaearon Wasn't sure what to do here. Should we show both warnings? How should they be formatted?

Comment (gaearon) [reply packages/react-reconciler/src/ReactDebugFiberPerf.js]: Maybe show more important one? There's not much screen space there.

**Comment by gaearon** [comment]
Is interrupting always bad?

**Comment by acdlite** [comment]
Not necessarily, though neither are cascading updates. Maybe we should add a different type of label formatted for non-scary information?

**Comment by acdlite** [comment]
An example of something that's always bad is expiration.

File: packages/react-reconciler/src/ReactDebugFiberPerf.js [hunk 2]
```diff
@@ -318,7 +318,8 @@ export function stopPhaseTimer(): void {
   }
 }
 
-export function startWorkLoopTimer(): void {
+export function startWorkLoopTimer(nextUnitOfWork: Fiber | null): void {
+  currentFiber = nextUnitOfWork;
```
Comment (gaearon) [inline packages/react-reconciler/src/ReactDebugFiberPerf.js]: This should be inside the `if (enableUserTimingAPI)` check.

Comment (acdlite) [reply packages/react-reconciler/src/ReactDebugFiberPerf.js]: Updated

**Pull request merged**